pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -228,6 +228,9 @@ func (m *kubeGenericRuntimeManager) startContainer(ctx context.Context, podSandb
 	}
 
 	target, err := spec.getTargetID(podStatus)
+	if restartCount > 0 {
+		m.recordContainerEvent(pod, container, "", v1.EventTypeNormal, events.RestartedContainer, fmt.Sprintf("Restarting container %s", container.Name))
```

Comment:
Since you already have the `restartCount`, do you want to use it in the event message? something like, "Restarting container %s, attempt %d"  ?

Reply:
Yes, that will be nice addition. I was wondering if we can add the reason behind the restart here. Im investigating on it but any hint will be really helpful.

Reply:
i found [this](https://github.com/kubernetes/kubernetes/blob/921fb0b7c5850db9b109f28dd669b81b1522e055/pkg/kubelet/kubelet_pods.go#L2038) interesting , i think we can add reason here if it is not mentioned. wdyt?

Reply:
Just wondering how that would look in the `kubectl` output  for the short-lived container with restart policy "Always" after thousands of restarts.

Reply:
> Just wondering how that would look in the `kubectl` output for the short-lived container with restart policy "Always" after thousands of restarts.

+1, sometimes our container may be restarted many times. If we send events in this way, there may be thousands of events in the cluster. This may not be what we want.

Reply:
How about if it is added on just first restart? we have a restart count so we can leverage that
Not mentioning restart again and again and only showing it on first restart , i think that will serve the purpose , WDYT?
@googs1025 @bart0sh

Reply:
@Ritikaa96 `Restart` is quite usual operation and it happens quite often, even for control plane containers. Publishing an event when restart happens first time wouldn't solve the issue as far as I understand. There should be a better way to signal about unexpectedly died container.

Reply:
I am lost at this point here , because mostly this state is included in `ContainerStateUnknown` in current [code](https://github.com/kubernetes/kubernetes/blob/cd04f47e8b287068e8bf0bfae6596b2606d54929/pkg/kubelet/container/runtime.go#L267C71-L267C71). I believe maybe we can include something there then?

Reply:
Yes, I think it makes sense to start caring of some of those events if we want to fix https://github.com/kubernetes/kubernetes/issues/123176

Reply:
I agree  with this. There are three states included in this in the comments in code, which are restarting, paused, dead

Reply:
Hi @bart0sh @seans3 @googs1025 @harche 

> Just wondering how that would look in the `kubectl` output for the short-lived container with restart policy "Always" after thousands of restarts.

regarding the above query , please see the screenshot below, as per these references I have restarted the container 10 times and the events are saved once while the repetition is saved only in the counter which seems efficient , so in case there is a pod which restart again and again, the events will not get crowded. i guess this solve this particular problem.
![Screenshot from 2024-11-27 17-22-58](https://github.com/user-attachments/assets/73c97cc0-4eb5-47e3-a936-9cb59f5701b8)
![Screenshot from 2024-11-27 17-21-58](https://github.com/user-attachments/assets/9ba3521f-93ed-4fd2-a607-bef901fbbb0e)